contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,365 @@
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            uint128 pcvDepositWeight = pcvDepositInfos[pcvDepositAddress].weight;
+
+            IPCVDepositAggregator(newAggregator).addPCVDeposit(pcvDepositAddress, pcvDepositWeight);
+        }
+
+        // Set all weights to zero (except for us)
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            pcvDepositInfos[pcvDepositAddress].weight = uint128(0);
+        }
+
+        // Rebalance (withdraws everything to us)
+        _rebalance();
+
+        // Send everything over to the new aggregator
+        IERC20(token).safeTransfer(address(newAggregator), IERC20(token).balanceOf(address(this)));
+
+        // Call rebalance on the new aggregator
+        IPCVDepositAggregator(newAggregator).rebalance();
+
+        // Remove all deposits.
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            _removePCVDeposit(pcvDepositAddress);
+        }
+
+        // Finally, set the new aggregator on the rewards asset manager itself
+        IRewardsAssetManager(rewardsAssetManager).setNewAggregator(address(newAggregator));
+    }
+
+    // ---------- View Functions ---------------
+    function resistantBalanceAndFei() external view virtual override returns (uint256, uint256) {
+        return (balance(), 0);
+    }
+
+    function balanceReportedIn() external view virtual override returns (address) {
+        return token;
+    }
+
+    function balance() public view virtual override returns (uint256) {
+        return getTotalBalance();
+    }
+
+    function percentHeld(address pcvDeposit, uint256 depositAmount) external view virtual override returns(Decimal.D256 memory) {
+        uint totalBalanceWithTheoreticalDeposit = getTotalBalance() + depositAmount;
+        uint targetBalanceWithTheoreticalDeposit = IPCVDeposit(pcvDeposit).balance() + depositAmount;
+
+        return Decimal.ratio(targetBalanceWithTheoreticalDeposit, totalBalanceWithTheoreticalDeposit);
+    }
+
+    function targetPercentHeld(address pcvDeposit) external view virtual override returns(Decimal.D256 memory) {
+        uint totalBalance = getTotalBalance();
+        uint targetBalance = IPCVDeposit(pcvDeposit).balance();
+
+        return Decimal.ratio(targetBalance, totalBalance);
+    }
+
+    function amountFromTarget(address pcvDeposit) external view virtual override returns(int256) {
+        uint totalBalance = getTotalBalance();
+
+        uint pcvDepositBalance = IPCVDeposit(pcvDeposit).balance();
+        uint pcvDepositWeight = pcvDepositInfos[address(pcvDeposit)].weight;
+
+        uint idealDepositBalance = pcvDepositWeight * totalBalance / totalWeight;
+        
+        return int(idealDepositBalance) - int(pcvDepositBalance);
```

Comment:
Thoughts on whether positive should mean above or below?

Reply:
Since we're using the nomenclature "amount [off] from target", a positive number should mean that we have overshot, eg we should take from this deposit. Likewise a negative number means that we are undershot and this deposit needs more.

...which makes my implementation backwards, but happy to correct.